### DISCUSSION TITLE
关于对某些第三方修改版的重要说明

### DISCUSSION THREAD

**Discussion by lisonge** [gkd-kit/gkd#679] [General]
**GKD 是开源的，你当然可以自由修改编译**

如果你只是自己编译给自己使用，没有发布/分享给大众，那没啥问题

---

但是如果你修改编译后**分享给其他用户**使用，那请你遵守以下几点要求

1. 请在应用内 **弹窗提示/其他醒目的区域/应用名称** 明确告知用户这是 GKD 的修改版，而不是官方版本，否则会给用户造成困惑
2. 请将应用内的 **反馈链接/官网链接/相关配置** 替换为你自己的配置，否则你的用户来找我反馈，这会给我增加不必要的烦恼

相关链接指代以下链接，请将这些链接替换为你自己的链接
- `gkd.li`, `*.gkd.li`
- `https://github.com/gkd-kit/`
- `https://registry.npmmirror.com/@gkd-kit/`

执行以上行为后

- 用户点击主页右上角的图标应该前往你的主页
- 用户在 设置-关于 点击的任意链接应该前往你的仓库，而不是 gkd 的官方仓库
- 相关的配置检测应该使用你配置的链接

**如果你做不到这些，那请你修改编译后不要分享给其他用户使用**

**Comment by aubur20240314** [comment]
为了更好的避免修改版找作者反馈造成麻烦，作者可尝试重构下代码（个人非 kotlin 开发者，暂时不能 fork 项目进行此重构）：
1、提取所有 gkd 官方配置到一个配置文件中；
2、应用各页面组件通过配置文件引入配置；
3、提交代码时提交空的配置文件；
4、构建时遇到空配置文件报错，强制需要填写配置文件；
5、README/文档网站中加上配置文件说明

**Reply by lisonge** [reply]
这纯粹是个人项目，也不对任何用户负责

代码风格完全看个人喜好

代码结构也并不是为修改版存在的

**Reply by aubur20240314** [reply]
理解👍，仅是提供一个更好保护自己的方式
